testing/policies/kyverno/kustomization.yaml

```diff
@@ -0,0 +1,37 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+  - https://github.com/kyverno/policies/pod-security/restricted
```

Comment:
❓ This seems to assert that all the containers are restricted to run as nonRoot OR the pod is restricted -- do I have that right?

Reply:
Also I think I need to brush up on Kyverno because I'm a little confused about this log:

```
    logger.go:42: 11:53:43 | security-context/10-10--kyverno | I0731 11:53:43.469837    7232 evaluate.go:63] EngineValidate "msg"="a condition failed in 'all' block" "kind"="StatefulSet" "name"="security-context-pgadmin" "namespace"="kuttl-test-select-rooster" "policy"="restrict-volume-types" "rule"="autogen-restricted-volumes" "condition"={"key":["name","emptyDir","name","persistentVolumeClaim","projected","name","emptyDir","name","emptyDir","name"],"operator":"AnyNotIn","value":["name","configMap","csi","downwardAPI","emptyDir","ephemeral","persistentVolumeClaim","projected","secret",""]}
    logger.go:42: 11:53:43 | security-context/10-10--kyverno | pass: 207, fail: 0, warn: 0, error: 0, skip: 830
```

A condition failed, but `fail: 0`... Is that what they're talking about when they log the note that I should log at level 5?

Reply:
> ❓ This seems to assert that all the containers are restricted to run as nonRoot OR the pod is restricted -- do I have that right?

Yep, that's right. [This policy](https://kyverno.io/policies/pod-security/restricted/require-run-as-nonroot/require-run-as-nonroot/) is also how they explain when/why to use [anyPattern](https://kyverno.io/docs/writing-policies/validate/#anypattern).

> I'm a little confused about this log

I noticed that, too, but haven't opened an issue. I looked some time ago, and my memory is that they added some logging deep somewhere that doesn't know when the expression is negated or not. 🤔 But I see `AnyNotIn` right there...

"Condition failed" is definitely confusing next to the overall "pass / fail" message.

Reply:
I mentioned it on Kyverno \#4074, and looks like they have an idea for a fix.